grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/utils/conversion/ConceptConverter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.reasoner.utils.conversion;
+
+import ai.grakn.concept.Concept;
+import ai.grakn.graql.Pattern;
+
+/**
+ * <p>
+ * Interface for conversion from {@link Concept}s.
+ * </p>
+ * @param <T> {@link Concept} type to convert from
+ *
+ * @author Kasper Piskorski
+ */
+public interface ConceptConverter<T extends Concept> {
+
+    Pattern pattern(T concept);
+
+    static Pattern toPattern(Concept concept) {
+        if (!concept.isThing()) return null;
```

Comment:
Instead of returning `null`, I'd recommend `throw`ing here.

But if there's a good reason to return null, make sure to mark the method as `@Nullable`.